plugins/funind/glob_termops.mli

```diff
@@ -122,3 +122,10 @@ val zeta_normalize : Glob_term.glob_constr -> Glob_term.glob_constr
 
 
 val expand_as : glob_constr -> glob_constr
+
+
+(* [resolve_and_replace_implicits ?expected_type env sigma rt] solves implicits of [rt] w.r.t. [env] and [sigma] and then replace them by their solution 
+ *)
+val resolve_and_replace_implicits :
```

Comment:
The point we dealing directly with `constr` and not `glob_constr` is that we would then work after case compilation which (potentially) generates many branches from only one branch in the user term (take for example `match n with 1000 => true | _ => false end` and thus add too many branches in the generated induction principle. I'll be very happy to deal directly with `constr` as soon as I will have a way to really decompile cases  at this level.